test/Streamly/Test/Data/Fold.hs

```diff
@@ -461,3 +480,4 @@ main = hspec $
         prop "distribute" Main.distribute
         prop "partition" Main.partition
         prop "unzip" Main.unzip
+        prop "foldIterate" Main.foldIterate
```

Comment:
The test should go in prelude tests, as the operation is in Prelude and not in Data.Fold.

Reply:
I've mimicked the parser layout. I remember seeing `parseIterate` and `parseMany` in `bench:Data.Parser`